nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/ir/OpDescriptorHolder.java

```diff
@@ -0,0 +1,99 @@
+/*
+ *  ******************************************************************************
+ *  *
+ *  *
+ *  * This program and the accompanying materials are made available under the
+ *  * terms of the Apache License, Version 2.0 which is available at
+ *  * https://www.apache.org/licenses/LICENSE-2.0.
+ *  *
+ *  *  See the NOTICE file distributed with this work for additional
+ *  *  information regarding copyright ownership.
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  * License for the specific language governing permissions and limitations
+ *  * under the License.
+ *  *
+ *  * SPDX-License-Identifier: Apache-2.0
+ *  *****************************************************************************
+ */
+package org.nd4j.ir;
+
+import lombok.val;
+import org.apache.commons.io.IOUtils;
+import org.nd4j.common.config.ND4JClassLoading;
+import org.nd4j.common.io.ClassPathResource;
+import org.nd4j.shade.protobuf.TextFormat;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.*;
+
+/**
+ * A utility class for accessing the nd4j op descriptors.
+ * May override default definition of {@link #nd4jFileNameTextDefault}
+ * with the system property {@link #nd4jFileSpecifierProperty}
+ * @author Adam Gibson
+ */
+public class OpDescriptorHolder {
+
+    public static String  nd4jFileNameTextDefault = "/nd4j-op-def.pbtxt";
```

Comment:
We've had issues with people creating uberjars and not packaging resources previously. So having something that is dependent on loading a resource may introduce new problems. 

In particular, Spring Boot has a special way of class loading and resource loading. I can imagine that it may break there.

Reply:
@treo in that case we can tell people to just override the classloader with their own. That functionality was merged a while back. I agree with you and will keep an eye on that though.